val.py

```diff
@@ -344,35 +435,114 @@ def run(
                 save_one_json(predn, jdict, path, class_map)  # append to COCO-JSON dictionary
             callbacks.run('on_val_image_end', pred, predn, path, names, im[si])
 
+            # TODO validate this line!
+            # TODO What does im contain
+            pred_clone[:, :4] = scale_boxes(im.shape[2:], pred_clone[:, :4],
+                                            im0[si].shape).round()  # TODO why not im[si].shape[2:]
+
+            image_filename, image_upload_date = parse_image_path(paths[si])
+
+            for *xyxy, conf, cls in pred_clone.tolist():
+                x1, y1 = int(xyxy[0]), int(xyxy[1])
+                x2, y2 = int(xyxy[2]), int(xyxy[3])
+                area_to_blur = im0[si][y1:y2, x1:x2]
+
+                blurred = cv2.GaussianBlur(area_to_blur, (135, 135), 0)
+                im0[si][y1:y2, x1:x2] = blurred
+
+                # Create an instance of DetectionInformation
+                detection_info = DetectionInformation(
+                    image_customer_name=customer_name,
+                    image_upload_date=image_upload_date,
+                    image_filename=image_filename,
+                    has_detection=True,
+                    class_id=int(cls),
+                    x_norm=x1,
+                    y_norm=y1,
+                    w_norm=x2,
+                    h_norm=y2,
+                    image_width=image_width,
+                    image_height=image_height,
+                    run_id=0
```

Comment:
@sdavrieux Now its hardcoded to 0. How do we define the run_id? Do we do it based on a unique number that we get from Azure ML? Or Unique number based on current date

Reply:
I suppose AzureML generates a string to identify each run, it would be nice to use that one so we keep everything "connectable".

Reply:
We can do using https://learn.microsoft.com/en-us/python/api/azureml-core/azureml.core.run(class)?view=azure-ml-py#azureml-core-run-get-details

from azureml.core import Run

# Get the current run context
run = Run.get_context()

# Get the Run ID using get_details()
run_id = run.get_details()['runId']
print("Run ID:", run_id)